doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -848,37 +1231,425 @@ The Ltac2 parser can be extended with syntactic notations.
 
-Note that abbreviations are not typechecked at all, and may result in typing
-errors after expansion.
+   Note that abbreviations are not type checked at all, and may result in typing
+   errors after expansion.
+
+.. _defining_tactics:
+
+Defining tactics
+~~~~~~~~~~~~~~~~
+
+All tactics used in Ltac2 must defined through notations.  These include built-in tactics (those
+defined in OCaml code in the |Coq| executable) and Ltac1 tactics, which are defined in `.v` files.
+
+Notations for many but not all built-in tactics are defined in `Notations.v`, which is automatically
+loaded with Ltac2.  The Ltac2 syntax for these tactics is often identical or very similar to the
+tactic syntax described in other chapters of this documentation.  These notations rely on tactic functions
+declared in `Std.v`.  Functions corresponding to some built-in tactics may not yet be defined in the
+|Coq| executable or declared in `Std.v`.  Adding them may require code changes to |Coq| or defining
+workarounds through Ltac1 (described below).
+
+Two examples of syntax differences:
+
+- There is no notation defined that's equivalent to :n:`intros until {| @ident | @num }`.  There is,
+  however, already an ``intros_until`` tactic function defined ``Std.v``, so it may be possible for a user
+  to add the necessary notation.
+- The build-in `simpl` tactic supports the use of scope keys in delta flags, e.g. :n:`simpl ["+"%nat]`
+  which is not accepted by Ltac2.  This is because Ltac2 uses a different
+  definition for :token:`delta_flag`; compare it to :token:`ltac2_delta_flag`.  This also affects
+  :tacn:`compute`.
+
+Ltac1 tactics are not automatically available in Ltac2.  (Note that some of the tactics described
+in the documentation are defined with Ltac1.)
+You can make them accessible in Ltac2 with the following commands:
+
+.. coqtop:: in
+
+   From Coq Require Import Lia.
+   Local Ltac2 lia_ltac1 () := ltac1:(lia).
+   Ltac2 Notation "lia" := lia_ltac1 ().
+
+A similar approach can be used to access missing built-in tactics.  See :ref:`simple_api` for an
+example that passes two parameters to a missing build-in tactic.
+
+.. _syntactic_classes:
+
+Syntactic classes
+~~~~~~~~~~~~~~~~~
+
+The simplest syntactic classes in Ltac2 notations represent individual nonterminals
+from the |Coq| grammar.  Only a few selected nonterminals are available as syntactic classes.
+In addition, there are metasyntactic operations for describing
+more complex syntax, such as making an item optional or representing a list of items.
+When parsing, each syntactic class expression returns a value that's bound to a name in the
+notation definition.
+
+Syntactic classes are described with a form of S-expression:
+
+   .. insertprodn ltac2_scope ltac2_scope
+
+   .. prodn::
+      ltac2_scope ::= @string
+      | @int
+      | @name
+      | @name ( {+, @ltac2_scope } )
+
+.. todo no syn class for ints or strings?
+  parm names are not reserved (e.g the var can be named "list1"
```

Comment:
```suggestion
  parm names are not reserved (e.g the var can be named "list1")
```

Reply:
Thanks.  FWIW, that's a non-user visible comment.